src/pykeen/models/base.py

```diff
@@ -32,27 +33,253 @@
 
 
 class Model(nn.Module, ABC):
-    """A base module for all of the KGE models."""
+    """A base module for KGE models.
+
+    Subclasses of :class:`Model` can decide however they want on how to store entities' and
+    relations' representations, how they want to be looked up, and how they should
+    be scored. The :class:`OModel` provides a commonly used interface for models storing entity
+    and relation representations in the form of :class:`pykeen.nn.Embedding`.
+    """
 
     #: Keep track of if this is a base model
     _is_base_model: ClassVar[bool]
 
     #: The default strategy for optimizing the model's hyper-parameters
     hpo_default: ClassVar[Mapping[str, Any]]
 
+    #: A triples factory with the training triples
+    triples_factory: TriplesFactory
+
+    #: The device on which this model and its submodules are stored
+    device: torch.device
+
+    _random_seed: Optional[int]
+
     #: The default loss function class
     loss_default: ClassVar[Type[Loss]] = MarginRankingLoss
     #: The default parameters for the default loss function class
     loss_default_kwargs: ClassVar[Optional[Mapping[str, Any]]] = dict(margin=1.0, reduction='mean')
     #: The instance of the loss
     loss: Loss
 
+    def __init_subclass__(cls, autoreset: bool = True, **kwargs):  # noqa:D105
+        cls._is_base_model = not autoreset
+        if not cls._is_base_model:
+            _add_post_reset_parameters(cls)
+
+    """Properties"""
+
+    @property
+    def can_slice_h(self) -> bool:
```

Comment:
while the new model throws away these checks completely since all models can be sliced, we can just override these functions to maintain backwards compatibility.